pkg/hsskey/filereader.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hsskey
+
+import (
+	"encoding/binary"
+	"fmt"
+	"hash/crc32"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+const (
+	baseSysfsPattern = "/sys/bus/i2c/devices/%s/eeprom"
+)
+
+// toOctalEscapeSequence converts a byte slice into similar format as C++ protobuf DebugString()
+// If the byte represents a printable ASCII character, add it as is to the output string
+// If it is non-printable or non-ASCII, add it as octal notation.
+func toOctalEscapeSequence(b []byte) string {
```

Comment:
consider commenting on why this is needed ?

Reply:
done